SofaKernel/framework/sofa/helper/io/MassSpringLoader.cpp

```diff
@@ -47,7 +48,155 @@ static void skipToEOL(FILE* f)
+//        file.close();
+//        return false;
+
+//    }
+
+//    file.close();
+//    return fileRead;
+//}
+
+
+
+//bool MeshXspLoader::readXsp (std::ifstream &file, bool vector_spring)
+//{
+//    dmsg_info() << "Reading Xsp file: " << vector_spring;
+
+//    std::string cmd;
+//    file >> cmd;
+
+//    // then find out number of masses and springs, not used.
+//    if (cmd == "numm")
+//    {
+//        int totalNumMasses = 0;
+//        file >> totalNumMasses;
+//    }
+
+//    if (cmd=="nums")
+//    {
+//        int totalNumSprings = 0;
+//        file >> totalNumSprings;
+//    }
+
+
+//    helper::vector<sofa::defaulttype::Vector3>& my_positions = *(d_positions.beginEdit());
+//    helper::vector<Edge >& my_edges = *(d_edges.beginEdit());
+
+//    helper::vector <Vector3>& my_gravity = *(gravity.beginEdit());
+//    helper::vector <double>& my_viscosity = *(viscosity.beginEdit());
+
+//    while (!file.eof())
+//    {
+//        file  >> cmd;
+//        if (cmd=="mass")
+//        {
+//            int index;
+//            char location;
+//            double px,py,pz,vx,vy,vz,mass=0.0,elastic=0.0;
+//            //bool fixed=false;
+//            file >> index >> location >> px >> py >> pz >> vx >> vy >> vz >> mass >> elastic;
+//            my_positions.push_back(Vector3(px, py, pz));
+//        }
+//        else if (cmd=="lspg")	// linear springs connector
+//        {
+//            int	index;
+//            Edge m;
+//            double ks = 0.0, kd = 0.0, initpos = -1;
+//            if (vector_spring)
+//            {
+//                double restx=0.0,resty=0.0,restz=0.0;
+//                file >> index >> m[0] >> m[1] >> ks >> kd >> initpos >> restx >> resty >> restz;
+//            }
+//            else
+//            {
+//                file >> index >> m[0] >> m[1] >> ks >> kd >> initpos;
+//            }
+//            --m[0];
+//            --m[1];
+
+//            addEdge(&my_edges, m);
+//        }
+//        else if (cmd == "grav")
+//        {
+//            double gx,gy,gz;
+//            file >> gx >> gy >> gz;
+//            my_gravity.push_back(Vector3(gx, gy, gz));
+//        }
+//        else if (cmd == "visc")
+//        {
+//            double visc;
+//            file >> visc;
+//            my_viscosity.push_back (visc);
+//        }
+//        else if (cmd == "step")
+//        {
+//        }
+//        else if (cmd == "frce")
+//        {
+//        }
+//        else if (cmd[0] == '#')	// it's a comment
+//        {
+//        }
+//        else		// it's an unknown keyword
+//        {
+//            msg_error() << "Unknown MassSpring keyword '" << cmd << "'.";
+//            d_positions.endEdit();
+//            d_edges.endEdit();
+//            gravity.endEdit();
+//            viscosity.endEdit();
+//            return false;
+//        }
+//    }
+
+//    d_positions.endEdit();
+//    d_edges.endEdit();
+//    gravity.endEdit();
+//    viscosity.endEdit();
+
+//    return true;
+//}
```

Comment:
That's a big block of commented code here! :open_mouth:

Reply:
Yes... this is the code made by Erik...in MeshXspLoader that I prefer to the "c" version that seems more complete.